src/java/org/apache/cassandra/transport/CQLMessageHandler.java

```diff
@@ -405,8 +463,40 @@ protected boolean processFirstFrameOfLargeMessage(IntactFrame frame, Limit endpo
             // max CQL message size defaults to 256mb, so should be safe to downcast
             int messageSize = Ints.checkedCast(header.bodySizeInBytes);
             receivedBytes += buf.remaining();
-
+            
             LargeMessage largeMessage = new LargeMessage(header);
+
+            if (DatabaseDescriptor.getNativeTransportRateLimitingEnabled())
+            {
+                long waitLengthMicros = requestRateLimiter.reserveAndGetWaitMicros();
```

Comment:
Shouldn't this be conditional also, for throwOnOverload? Otherwise we may find ourselves trapped in a spiral of discarding messages.

Reply:
Agreed. Should be resolved [here|https://github.com/apache/cassandra/pull/1045/commits/19b9d272383da4b01e41f401ef3156954f01934a].